query/PRoST-query/src/main/java/translator/Translator.java

```diff
@@ -3,9 +3,7 @@
 import java.util.ArrayDeque;
```

Comment:
I don't think the source code would be the ideal place how we create the tree. There's already a paper in the readme that explains how we create a tree with WPT and VP. 
We can index the report about with the explanation and evaluation of the IWPT. 
JWPT should be explained in the next paper if we are going to evaluate it too.

Reply:
The comments added to the source code have to explain it and if that's not the best place where you can explain what your code is doing I do not know which is. For example, I didn't understaitially hond inw we chose which triples to group into jwpt node and which to vp node. Therefore, what I meant is "Please comment your code". Of course, you can add links to the papers. But apart from that you also need to add some comments that are documenting your code. I haven't said that you need to write an essay. Two, three sentences for a table would be enough.

Reply:
A JWPT node is created if a group is larger than the minimum group size which is given as an input parameter, otherwise VP nodes are created otherwise.

```
if (joinedGroup.size()>=minimumGroupSize){
	nodesQueue.add(new JptNode(joinedGroup, prefixes));
} else {
	createVpNodes(new ArrayList<>(joinedGroup.getWptGroup()), nodesQueue);
```

I took care to document most methods. Which ones exactly are you having trouble with?